test/import-style.js

```diff
@@ -0,0 +1,603 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import {outdent} from 'outdent';
+import rule from '../rules/import-style';
+import visualizeRuleTester from './utils/visualize-rule-tester';
+
+const ruleTester = avaRuleTester(test, {
+	parserOptions: {
+		sourceType: 'module',
+		ecmaVersion: 2020
+	}
+});
+
+const options = {
+	checkExportFrom: true,
+	styles: {
+		unassigned: {
+			unassigned: true
+		},
+		default: {
+			default: true
+		},
+		namespace: {
+			namespace: true
+		},
+		named: {
+			named: true
+		}
+	}
+};
+
+const unassignedError = {
+	messageId: 'importStyle',
+	data: {
+		allowedStyles: 'unassigned',
+		moduleName: 'unassigned'
+	}
+};
+
+const defaultError = {
+	messageId: 'importStyle',
+	data: {
+		allowedStyles: 'default',
+		moduleName: 'default'
+	}
+};
+
+const namespaceError = {
+	messageId: 'importStyle',
+	data: {
+		allowedStyles: 'namespace',
+		moduleName: 'namespace'
+	}
+};
+
+const namedError = {
+	messageId: 'importStyle',
+	data: {
+		allowedStyles: 'named',
+		moduleName: 'named'
+	}
+};
+
+const addDefaultOptions = test => {
+	if (typeof test === 'string') {
+		test = {
+			code: test
+		};
+	}
+
+	return {
+		options: [options],
+		...test
+	};
+};
+
+ruleTester.run('import-style', rule, {
+	valid: [
+		'require(\'unassigned\')',
+		'const {} = require(\'unassigned\')',
+		'import \'unassigned\'',
+		'import {} from \'unassigned\'',
+		'import(\'unassigned\')',
+		'export {} from \'unassigned\'',
+
+		'const x = require(\'default\')',
+		'const {default: x} = require(\'default\')',
+		'import x from \'default\'',
+		outdent`
+			async () => {
+				const {default: x} = await import('default');
+			}
+		`,
+		'export {default} from \'default\'',
+
+		'const x = require(\'namespace\')',
+		'import * as x from \'namespace\'',
+		outdent`
+			async () => {
+				const x = await import('namespace');
+			}
+		`,
+		'export * from \'namespace\'',
+
+		'const {x} = require(\'named\')',
```

Comment:
Do we support check `const x = require('named').x`?

Reply:
I added the test as `valid`. I hope people use the `prefer-destructuring` rule.